lib/fs/filec.mli

```diff
@@ -7,6 +7,10 @@ type t =
       offset : int;
     }
 
+type file_type =
+  | BinaryFile
+  | TextFile
```

Comment:
Since we have this type now, let's change `t` here from a sum type to a simple record having `lines` and `offset`.

For binary files, we'll just create a record with `offset = 0` and lines being just `[| Pretty.Doc.str "This file is binary and cannot be displayed" |]` (we can still keep a single global variable for maximum sharing).